xcodeproj/internal/bazel_labels.bzl

```diff
@@ -0,0 +1,106 @@
+"""API for resolving and managing Bazel labels"""
```

Comment:
Did `Label()` not work as is?

Reply:
The `Label()` only accepts absolute labels. The `bazel_labels.normalize()` does the work of converting an absolute or relative label to a fully-qualified, absolute label. This allows a client to use relative labels. If we were not sharing the schemes/target list for `//tools/generator:xcodeproj`, we could declare the schemes in the `BUILD` file like so:

```python
_SCHEMES = [
    xcode_schemes.scheme(
        name = "Generator",
        launch_action = xcode_schemes.launch_action(":generator"),
        test_action = xcode_schemes.test_action([
            "//tools/generator/test:tests",
        ]),
    ),
]
```

I added a TODO to #573 to add the validity check that we discussed. I plan to add a `bazel_labels.validate()` that will confirm that a label points to something real.